salt/client/ssh/client.py

```diff
@@ -196,3 +197,42 @@ def cmd_async(self, low, timeout=None):
         '''
         # TODO Not implemented
         raise SaltClientError
+
+    def cmd_subset(
+            self,
+            tgt,
+            fun,
+            arg=(),
+            timeout=None,
+            tgt_type='glob',
+            ret='',
+            kwarg=None,
+            sub=3,
+            **kwargs):
+        '''
+        Execute a single command via the salt-ssh subsystem on a subset of the targeted minions
+
+        .. versionadded:: Nitrogen
+        '''
+        if 'expr_form' in kwargs:
+            salt.utils.warn_until(
+                'Fluorine',
+                'The target type should be passed using the \'tgt_type\' '
+                'argument instead of \'expr_form\'. Support for using '
+                '\'expr_form\' will be removed in Salt Fluorine.'
+            )
+            tgt_type = kwargs.pop('expr_form')
+        minion_ret = self.cmd(tgt,
+                              'sys.list_functions',
+                              tgt_type=tgt_type,
+                              **kwargs)
+        minions = list(minion_ret)
+        random.shuffle(minions)
+        f_tgt = []
+        for minion in minions:
+            if fun in minion_ret[minion]['return']:
+                f_tgt.append(minion)
+            if len(f_tgt) >= sub:
+                break
+        ret = self.cmd(f_tgt, fun, arg, timeout, tgt_type='list', ret=ret, kwarg=kwarg, **kwargs)
+        return [ret]
```

Comment:
Can someone help me figure out why this has to be a list?

Maybe it is because the return from LocalClient.cmd() is an ordered dictionary?

Thanks,
Daniel